modules/apps/static/portal-file-install/portal-file-install-impl/src/main/java/com/liferay/portal/file/install/internal/configuration/ConfigurationFileInstaller.java

```diff
@@ -265,6 +287,51 @@ StringPool.EQUAL, _escapeFilterValue(fileName),
 		return null;
 	}
 
+	private Long _getCompanyId(
+		ExtendedObjectClassDefinition.Scope scope,
+		Dictionary<String, Object> dictionary, String fileName) {
+
+		if (!DBPartition.isPartitionEnabled() ||
+			(scope != ExtendedObjectClassDefinition.Scope.COMPANY) ||
+			(dictionary == null)) {
+
+			return 0L;
+		}
+
+		Long companyId = (Long)dictionary.get(scope.getPropertyKey());
+
+		if (companyId != null) {
+			if (!ArrayUtil.contains(
+					PortalInstancePool.getCompanyIds(), companyId)) {
```

Comment:
@achaparro @LuisOrtiz89 configuration file processing logic starts way earlier than PortalInstancePool initialization, you may not see the existing company and have a false complain here. And eventually saves the configuration into the wrong db partition.

Reply:
got it @shuyangzhou, I will change it to use something reliable at this point